common/Environments/Models/ComputeSystemCache.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Threading.Tasks;
+using DevHome.Common.Environments.Helpers;
+using DevHome.Common.Helpers;
+using Microsoft.Windows.DevHome.SDK;
+using Serilog;
+using Windows.Foundation;
+
+namespace DevHome.Common.Environments.Models;
+
+/// <summary>
+/// Wrapper class for the IComputeSystem interface that can be used throughout the application.
+/// It uses lazy initialization to cache the results of OOP calls to the extension methods.
+/// The implementation of the calls to IComputeSystem remains in ComputeSystem class for now.
+/// Once we remove direct usage of ComputeSystem class throughout the code, we can merge these two classes.

```

Comment:
Is there an issue to track this?

Reply:
No, I don't think we want an issue for every refactoring opportunity. We'll replace it gradually while working on the code. In some places it can be less trivial as we'll need fresh data instead of data from the cache, so the functionality will need to be extended with proper testing.